### PR TITLE
Add comprehensive terminal graphics protocol support with runtime detection

### PR DESCRIPTION
**Pull request by gnodet**
## 🎨 **Terminal Graphics Protocol Support**

This PR adds comprehensive support for terminal graphics protocols with intelligent runtime detection, making JLine capable of displaying images in modern terminals.

### **1. Multi-Protocol Graphics Support**
- **Kitty Graphics Protocol** (Priority: 90) - Modern, feature-rich protocol
- **iTerm2 Inline Images** (Priority: 70) - iTerm2's proprietary protocol  
- **Sixel Graphics** (Priority: 10) - Widely supported legacy protocol

### **2. Intelligent Runtime Detection**
- **Sixel Detection**: Device Attributes query (`ESC[c`) - same method as `lsix` command
- **Kitty Detection**: Official Kitty graphics protocol query with quiet mode
- **Automatic Fallback**: Falls back to static detection if runtime detection fails
- **Performance Optimized**: Skips runtime detection for known non-supporting terminals

### **3. Comprehensive Terminal Support**

| Terminal | Sixel | Kitty | iTerm2 | Detection Method |
|----------|-------|-------|--------|------------------|
| **Kitty** | ❌ No | ✅ Yes | ❌ No | Runtime + Static |
| **Ghostty** | ❌ No | ✅ Yes | ❌ No | Static (confirmed by maintainer) |
| **iTerm2** | ✅ Yes | ✅ Yes | ✅ Yes | Static + Environment |
| **WezTerm** | ✅ Yes | ✅ Yes | ❌ No | Static + Environment |
| **xterm** | ✅ Yes | ❌ No | ❌ No | Runtime + Static |
| **foot** | ✅ Yes | ❌ No | ❌ No | Runtime + Static |
| **Konsole** | ✅ Yes | ❌ No | ❌ No | Runtime + Static |

### **Core Classes**
- `TerminalGraphicsManager` - Unified interface for graphics operations
- `SixelGraphics` - Sixel protocol implementation with runtime detection
- `KittyGraphics` - Kitty protocol implementation  
- `ITerm2Graphics` - iTerm2 protocol implementation
- `TerminalGraphics` - Common interface for all protocols

### **Runtime Detection Logic**
```java
// Sixel Detection (Device Attributes)
terminal.writer().print("\033[c");
// Response: ESC[?1;2;4;6;9;15;18;21;22c
// Code "4" indicates Sixel support

// Kitty Detection (Graphics Protocol Query)
terminal.writer().print("\033_Ga=q,s=1,v=1,i=1,I=1,q=2\033\\");
// Response: ESC_Gi=1,I=1;OKESC\ (if supported)
```

### **Usage Examples**
```java
// Automatic protocol selection
TerminalGraphicsManager.displayImage(terminal, bufferedImage);

// With options
TerminalGraphics.ImageOptions options = new TerminalGraphics.ImageOptions()
    .width(100).height(100).preserveAspectRatio(true);
TerminalGraphicsManager.displayImage(terminal, image, options);

// Force specific protocol
TerminalGraphicsManager.forceProtocol(TerminalGraphics.Protocol.KITTY);

// Check support
boolean supported = TerminalGraphicsManager.isGraphicsSupported(terminal);
Optional<TerminalGraphics> best = TerminalGraphicsManager.getBestProtocol(terminal);
```

### **New Test Files** (846+ lines of tests)
- `RuntimeDetectionTest.java` - Runtime detection functionality
- `KittyGraphicsTest.java` - Kitty protocol comprehensive tests
- `GraphicsErrorHandlingTest.java` - Edge cases and error scenarios
- `GraphicsIntegrationTest.java` - End-to-end integration tests
- Enhanced `TerminalGraphicsTest.java` - Manager functionality

### **Test Coverage Areas**
✅ **Runtime Detection** - Device queries, timeouts, error handling  
✅ **Protocol Selection** - Priority ordering, terminal-specific logic  
✅ **Image Processing** - Multiple formats, color depths, resizing  
✅ **Error Handling** - I/O errors, corrupted data, memory constraints  
✅ **Integration** - End-to-end workflows, file operations  
✅ **Concurrency** - Thread safety, concurrent access  

## 🎯 **Key Benefits**

1. **✅ Future-Proof**: Works with unknown terminals that support protocols
2. **✅ Accurate Detection**: Runtime queries provide definitive answers
3. **✅ Performance**: Smart detection avoids unnecessary queries
4. **✅ Robust**: Comprehensive error handling and fallbacks
5. **✅ Compatible**: Maintains backward compatibility
6. **✅ Extensible**: Easy to add new protocols via ServiceLoader

## 🔍 **Testing**

```bash
# Run all graphics tests
./mvnw test -pl terminal -Dtest="*Graphics*Test"

# Test with example
./build example TerminalGraphicsExample -- --list-protocols
./build example TerminalGraphicsExample -- --display-image path/to/image.png
```

## 📚 **Documentation**

- Complete JavaDoc documentation for all classes
- Usage examples in `TerminalGraphicsExample`
- Comprehensive `SIXEL_GRAPHICS.md` guide

## 🐛 **Fixes**

- **Ghostty Hanging Issue**: Fixed runtime detection causing hangs in Ghostty
- **Response Leakage**: Prevented terminal responses from appearing in output
- **Protocol Priority**: Ensured correct protocol selection based on terminal capabilities
- **Memory Management**: Proper image resizing and memory handling

## 🔄 **Backward Compatibility**

This PR maintains full backward compatibility. All existing functionality continues to work unchanged, with graphics support being purely additive.

---

**Ready for review!** This implementation brings JLine's graphics capabilities to modern standards while maintaining the library's reliability and performance characteristics. 🚀✨

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) with guidance from the PR author

### CODE REVIEW

File: terminal/src/main/java/org/jline/terminal/impl/SixelGraphics.java [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+
+            // Read response with configurable timeout (default: 200ms for faster response)
+            long timeoutMs = Long.parseLong(System.getProperty(GRAPHICS_SIXEL_TIMEOUT, "200"));
+            long subsequentTimeoutMs = Long.parseLong(System.getProperty(GRAPHICS_SIXEL_SUBSEQUENT_TIMEOUT, "25"));
+            String response = readTerminalResponse(terminal, timeoutMs, subsequentTimeoutMs);
+            if (response != null) {
+                // Look for code "4" which indicates Sixel graphics support
+                // Response format: ESC[?1;2;4;6;9;15;18;21;22c
+                // Code "4" = Sixel graphics support
+                return response.contains(";4;") || response.contains(";4c") || response.endsWith("4c");
```
Comment (j4james) [inline terminal/src/main/java/org/jline/terminal/impl/SixelGraphics.java]: That last test - `endsWith("4c")` - will match extensions `14`, `24`, `34`, etc. none of which are Sixel. It's not needed here.

File: terminal/src/main/java/org/jline/terminal/impl/SixelGraphics.java [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+        }
+
+        // Check for XTerm with sixel support
+        if (termEnv != null && termEnv.startsWith("xterm")) {
+            // XTerm supports sixel since patch #359
+            // Unfortunately, there's no reliable way to detect the patch level
+            // from environment variables alone
+            // For now, assume modern xterm installations support sixel
+            return true;
+        }
```
Comment (j4james) [inline terminal/src/main/java/org/jline/terminal/impl/SixelGraphics.java]: Note that there are lots of terminals that set their `TERM` environment variable to some variant of `xterm`, and they won't all support sixel. Even xterm doesn't support it by default as far as I can recall (you need to launch it with `-ti vt340`, or set an appropriate resource setting). So this test is likely to produce false positives in a lot of cases. I don't know if that's a concern or not.

File: terminal/src/main/java/org/jline/terminal/impl/SixelGraphics.java [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+     */
+    private static String generateSixelData(BufferedImage image) throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        // Start with DCS sequence and Sixel introduction
+        // Parameters: 0;1;q means:
+        // - 0: Pixel aspect ratio 1:1
+        // - 1: Background color is set to black
+        // - q: Sixel mode
+        baos.write((DCS + SIXEL_INTRO).getBytes(StandardCharsets.US_ASCII));
```
Comment (j4james) [inline terminal/src/main/java/org/jline/terminal/impl/SixelGraphics.java:427]: That first `0` is actually requesting a `2:1` aspect ratio, unless you're overriding it elsewhere with a raster attributes command. So if you want `1:1`, the `SIXEL_INTRO` should be something like `9;1q`. Most terminals ignore aspect ratios, so it may seem like it's working correctly, but the images will end up stretched on terminals that do support them.
